pkg/maps/metricsmap/metricsmap.go

```diff
@@ -223,20 +223,29 @@ func updatePrometheusMetrics(key *Key, val *Value) {
 func SyncMetricsMap(ctx context.Context) error {
 	entry := make([]Value, possibleCpus)
 	file := bpf.MapPath(MapName)
-	metricsmap, err := bpf.OpenMap(file)
 
-	if err != nil {
-		return fmt.Errorf("unable to open metrics map: %s", err)
+	var err error
+	metricsMap := bpf.GetMap(file)
+	if metricsMap == nil {
+		// Open the map and leave it open, since SyncMetricsMap is called
+		// periodically and it makes sense to use an already opened map rather
+		// than opening the map again and again.
+		// This also prevents the constant registration and unregistration of the
+		// Map.
+		metricsMap, err = bpf.OpenMap(file)
+
+		if err != nil {
+			return fmt.Errorf("Unable to open metrics map: %s", err)
+		}
 	}
-	defer metricsmap.Close()
```

Comment:
LGTM with a minor concern: so the map never gets closed, I wonder if that will have any consequences. @brb WDYT ?

Reply:
Good point, but AFAIK the map fd should get closed by the kernel when a cilium-agent process terminates. /cc @joestringer